rl4j/rl4j-core/src/main/java/org/deeplearning4j/rl4j/experience/ExperienceHandler.java

```diff
@@ -0,0 +1,49 @@
+/*******************************************************************************
+ * Copyright (c) 2020 Konduit K.K.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Apache License, Version 2.0 which is available at
+ * https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ******************************************************************************/
+package org.deeplearning4j.rl4j.experience;
+
+import org.deeplearning4j.rl4j.observation.Observation;
+
+import java.util.List;
+
+/**
+ * A common interface to all classes capable of handling experience generated by the agents in a learning context.
+ *
+ * @param <A> Action type
+ * @param <E> Experience type
+ *
+ * @author Alexandre Boulanger
+ */
+public interface ExperienceHandler<A, E> {
+    void addExperience(Observation observation, A action, double reward, boolean isTerminal);
+
+    /**
+     * Called when the episode is done with the last observation
+     * @param observation
+     */
+    void setFinalObservation(Observation observation);
+
+    /**
+     * The elements are returned in the historical order (i.e. in the order they happened)
+     * @return The list of experience elements
+     */
+    List<E> generateTrainingBatch();

```

Comment:
I think you said you added this to my other branch, but a function to get a list of all the training experience or at least the current size of the experience buffer would be good

Reply:
I added `getTrainingBatchSize()` that returns the size of the batch that `generateTrainingBatch()` would return.